src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1518 @@
+     }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code byte} after
+     * a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public byte byteValue() {
+        return (byte)floatValue();
+    }
+
+    /**
+     * {@return a string representation of this {@code Float16}}
+     *
+     * @implSpec
+     * This method returns the result of {@code Float16.toString(this)}.
+     */
+    public String toString() {
+        return toString(this);
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code short}
+     * after a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public short shortValue() {
+        return (short)floatValue();
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as an {@code int} after
+     * a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public int intValue() {
+        return (int)floatValue();
+    }
+
+    /**
+     * {@return value of this {@code Float16} as a {@code long} after a
+     * narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public long longValue() {
+        return (long)floatValue();
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code float}
+     * after a widening primitive conversion}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @jls 5.1.2 Widening Primitive Conversion
+     */
+    @Override
+    public float floatValue() {
+        return float16ToFloat(value);
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code double}
+     * after a widening primitive conversion}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @jls 5.1.2 Widening Primitive Conversion
+     */
+    @Override
+    public double doubleValue() {
+        return (double)floatValue();
+    }
+
+    /**
+     * {@return a hash code for this {@code Float16} object}
+     *
+     * The general contract of {@code Object#hashCode()}is satisfied;
+     * all NaN values have the same hash value.
+     */
+    @Override
+    public int hashCode() {
+        return hashCode(this);
+    }
+
+    /**
+     * Returns a hash code for a {@code Float16} value; compatible with
+     * {@code Float16.hashCode()}.
+     *
+     * @param value the value to hash
+     * @return a hash code value for a {@code Float16} value.
+     */
+    public static int hashCode(Float16 value) {
+        Float16 f16 = isNaN(value) ? Float16.NaN : value;
+        return (int)float16ToRawShortBits(f16);
```

Comment:
What about randomizing the returned value? IIUC, JEP 401 does not speak against.
The purpose would not be to conceal anything, but to ensure that clients cannot rely on specific values.

```
    private static final int HASH_MASK = RandomGenerator.getDefault().nextInt();
    
    /**
     * Returns a hash code for a {@code Float16} value; compatible with
     * {@code Float16.hashCode()}.
     *
     * @param value the value to hash
     * @return a hash code value for a {@code Float16} value.
     */
    public static int hashCode(Float16 value) {
        Float16 f16 = isNaN(value) ? Float16.NaN : value;
        int bits = float16ToRawShortBits(f16);
        return (bits << 16 | bits & 0xFFFF) ^ HASH_MASK;
    }
```

Reply:
Hmm. Certainly at this point, I don't think hash randomization is necessary; although it might be helpful at a later point. Set.of() uses randomized iteration order as a precedent in that direction.